### PR TITLE
agent: retry instrumentation on errors

### PR DESCRIPTION
**Pull request by bertschneider**
Errors retransforming a class should not prevent Jazzer from starting. These classes are assumed to be not interesting nor important for the fuzzing run and can be safely ignored.

### CODE REVIEW

**Comment by bertschneider** [comment]
Resolves a part of #648.

**Comment by davidvoit** [comment]
Is the vararg retransformation really that much faster than just doing a for loop with a single class here?
But i must confess the binary search way is quite clever.

**Comment by bertschneider** [comment]
I didn't measure it, but using a simple loop (as I did in the beginning) seemed to take reallllly long.

**Comment by davidvoit** [comment]
Just checked the openjdk code. This is really not just convenient API stuff. The varargs classes are called in jni code as array parameter.
That is non jit land, and so your approach seems to be the most sensible one.

**Review by fmeum — APPROVED** [review]

File: src/main/java/com/code_intelligence/jazzer/agent/Agent.kt [hunk 1]
```diff
@@ -145,10 +145,22 @@ fun installInternal(
 
     if (classesToRetransform.isNotEmpty()) {
         if (instrumentation.isRetransformClassesSupported) {
-            instrumentation.retransformClasses(*classesToRetransform)
+            retransformClasses(instrumentation, classesToRetransform)
+        }
+    }
+}
+
+private fun retransformClasses(instrumentation: Instrumentation, classesToRetransform: Array<Class<*>>) {
```
[the remark this answers is not in the document]
Comment (bertschneider) [reply src/main/java/com/code_intelligence/jazzer/agent/Agent.kt]: Good point. Done.

**Pull request merged**